profile/encode.go

```diff
@@ -304,6 +320,12 @@ func (p *Profile) postDecode() error {
 		}
 		if len(numLabels) > 0 {
 			s.NumLabel = numLabels
+			for key, units := range numUnits {
+				if lenUnits, lenLabels := len(units), len(numLabels[key]); lenUnits > 0 && lenUnits < lenLabels {
```

Comment:
You don't need the "&& lenUnits < lenLabels" part I think since the condition is checked in padStringArray anyway.